zone/embparser.cpp

```diff
@@ -1270,25 +1270,23 @@ void PerlembParser::ExportItemVariables(std::string &package_name, Mob *mob)
 		perl->eval(std::string("%").append(hashname).append(" = ();").c_str());
 
 		for (int slot = EQ::invslot::EQUIPMENT_BEGIN; slot <= EQ::invslot::GENERAL_END; slot++) {
-			char *hi_decl = nullptr;
 			int  itemid   = mob->CastToClient()->GetItemIDAt(slot);
 			if (itemid != -1 && itemid != 0) {
-				MakeAnyLenString(&hi_decl, "push (@{$%s{%d}},%d);", hashname.c_str(), itemid, slot);
-				perl->eval(hi_decl);
-				safe_delete_array(hi_decl);
+				// this is really ugly with fmtlib, I think I did it right
+				auto hi_decl = fmt::format("push (@{{${0}{{{1}}}}},{2});", hashname, itemid, slot);
```

Comment:
Bet this was fun, looks right though

Reply:
Gonna not resolve this for now, would like some others to double check it as well.